heat/core/tests/test_suites/basic_test.py

```diff
@@ -0,0 +1,173 @@
+from unittest import TestCase
+from heat.core import dndarray, MPICommunication, MPI, types, factories
+import heat as ht
+
+import numpy as np
+import torch
+
+
+class BasicTest(TestCase):
+
+    __comm = MPICommunication()
+    __device = None
+
+    @property
+    def comm(self):
+        return BasicTest.__comm
+
+    @property
+    def device(self):
+        return BasicTest.__device
+
+    def get_rank(self):
+        return self.comm.rank
+
+    def get_size(self):
+        return self.comm.size
+
+    def assert_array_equal(self, heat_array: dndarray.DNDarray, numpy_array):
+        """
+        Check if the heat_array is equivalent to the numpy_array. Therefore first the split heat_array is compared to
+        the corresponding numpy_array slice locally and second the heat_array is combined and fully compared with the
+        numpy_array.
+        Note if the heat array is split it also needs to be balanced.
+
+        Parameters
+        ----------
+        heat_array: heat.DNDarray
+            The heat array which should be checked.
+        numpy_array: numpy.ndarray
+            The numpy array against which the heat_array should be checked.
+
+        Raises
+        ------
+        AssertionError if the arrays do not equal.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import heat as ht
+        >>> a = ht.ones((5, 5), split=1, dtype=ht.int32)
+        >>> b = np.ones((5, 5), dtype=np.int32)
+        >>> self.assert_array_equal(a, b)
+
+        >>> c = np.ones((5, 5), dtype=np.int64)
+        >>> self.assert_array_equal(a, c)
+        AssertionError: [...]
+        >>> c = np.zeros((5, 5), dtype=np.int32)
+        >>> self.assert_array_equal(a, c)
+        AssertionError: [...]
+        """
+        # Determine with what kind of numbers we are working
+        compare_func = np.array_equal if np.issubdtype(numpy_array.dtype, np.integer) else np.allclose
+
+        self.assertIsInstance(heat_array, dndarray.DNDarray,
+                              'The array to test was not a instance of ht.DNDarray. '
+                              'Instead got {}.'.format(type(heat_array)))
+        self.assertIsInstance(numpy_array, np.ndarray,
+                              'The array to test against was not a instance of numpy.ndarray. '
+                              'Instead got {}.'.format(type(numpy_array)))
+        self.assertEqual(heat_array.shape, numpy_array.shape,
+                         'Global shapes do not match. Got {} expected {}'.format(heat_array.shape, numpy_array.shape))
+        split = heat_array.split
+        offset, local_shape, slices = heat_array.comm.chunk(heat_array.gshape, split)
```

Comment:
comm checks should happen here as well

Reply:
For what should the comm be checked? That he is present and correct?